solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,29 +19,56 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
+//        throw new UnsupportedOperationException("Implement me!");
     }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не должно быть null");
+        Objects.requireNonNull(gameState, "Сохраняемый объект не должен быть null");
+
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + File.separator + filename);
+             ObjectOutputStream outputStream = new ObjectOutputStream(fos)) {
+
+            outputStream.writeObject(gameState);
+
+        } catch (FileNotFoundException e) {
+            throw new SaveGameException("Файл не найден", e, SaveGameException.Type.USER, gameState);
+        } catch (IOException e) {
+            throw new SaveGameException("Возникла ошибка при записи объекта в поток", e, SaveGameException.Type.IO, gameState);
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public MapState<GameObject> loadGame(String filename) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не должно быть null");
+        MapState<GameObject> savable = null;
+
+        try (FileInputStream fis = new FileInputStream(filesDirectory + File.separator + filename);
+             ObjectInputStream inputStream = new ObjectInputStream(fis)) {
+
+            savable = (MapState<GameObject>) inputStream.readObject();
+
+        } catch (FileNotFoundException e) {
+            throw new SaveGameException("Файл не найден", e, SaveGameException.Type.USER, savable);
+        } catch (IOException e) {
+            throw new SaveGameException("Возникла ошибка при чтении объекта из потока", e, SaveGameException.Type.IO, savable);
```

Comment:
Line is longer than 120 characters (found 127).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-786947779>*